File: src/js/src/module/ModuleEvent.js
/**
 * 模块事件
 * @type {[type]}
 * add by movinliao
 */
var BaseEvent = require('../base/BaseEvent');

var ModuleEvent = BaseEvent.extend(function () {
  this.priority = 0;
});

/**
 *日志
 */
BaseEvent.LOG = 'moduleevent.log';
/**
 *模块初始化
 */
BaseEvent.INIT = 'moduleevent.init';
/**
 *模块销毁
 */
BaseEvent.DESTROY = 'moduleevent.destroy';
/**
 *模块开始
 */
BaseEvent.START = 'moduleevent.start';
/**
 *模块停止
 */
BaseEvent.STOP = 'moduleevent.stop';
/**
 *块加入
 */
BaseEvent.ADD = 'moduleevent.add';
/**
 *块移除
 */
BaseEvent.REMOVE = 'moduleevent.remove';

module.exports = ModuleEvent;

File: src/android/ModuleEvent.java
package com.duowan.yylove.base;

import com.duowan.yylove.base.events.BaseEvent;
import com.duowan.yylove.base.events.EventKey;

/**
 * 功能：模块事件定义
 * Created by movinliao on 2018/4/28.
 */
public class ModuleEvent extends BaseEvent {
    public int priority = 0;

    @EventKey
    public static final String kLog = "ModuleEvent.kLog";
    @EventKey
    public static final String kInit = "ModuleEvent.kInit";
    @EventKey
    public static final String kDestroy = "ModuleEvent.kDestroy";
    @EventKey
    public static final String kStart = "ModuleEvent.kStart";
    @EventKey
    public static final String kStop = "ModuleEvent.kStop";
    @EventKey
    public static final String kAdd = "ModuleEvent.kAdd";
    @EventKey
    public static final String kRemove = "ModuleEvent.kRemove";

    public ModuleEvent(String key, Object data) {
        super(key, data);
    }

    public ModuleEvent(String key, Object data, int priority) {
        super(key, data);
        this.priority = priority;
    }
}
